cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -321,11 +325,40 @@ func (m *gceManagerImpl) forceRefresh() error {
 		klog.Errorf("Failed to fetch MIGs: %v", err)
 		return err
 	}
+	m.refreshAutoscalingOptions()
 	m.lastRefresh = time.Now()
 	klog.V(2).Infof("Refreshed GCE resources, next refresh after %v", m.lastRefresh.Add(refreshInterval))
 	return nil
 }
 
+func (m *gceManagerImpl) refreshAutoscalingOptions() {
+	for _, mig := range m.cache.GetMigs() {
+		template, err := m.migInstanceTemplatesProvider.GetMigInstanceTemplate(mig.GceRef())
+		if err != nil {
+			klog.Warningf("Not evaluating autoscaling options for %q MIG: failed to find corresponding instance template", mig.GceRef(), err)
+			continue
+		}
+		if template.Properties == nil {
+			klog.Warningf("Failed to extract autoscaling options from %q metadata: instance template is incomplete", template.Name)
+			continue
+		}
+		kubeEnvValue, err := getKubeEnvValueFromTemplateMetadata(template)
+		if err != nil {
+			klog.Warningf("Failed to extract KubeEnv from %q instance template's metadata: %v", template.Name, err)
```

Comment:
nit: this is a very generic warning (we get a lot of stuff from kube env in template). I like the ones above more, as they make it clear where the error is coming from.

Reply:
Right, updated to make it clear this comes from extracting autoscaling options.